src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -140,7 +140,27 @@ def _bigquery(self) -> bigquery.Client:
             return bigquery.Client(project=self._source.project)
 
     def _run_sql(self, query: str) -> BigQueryRows:
-        return self._bigquery.query(query)
+        attempt = 0
+        backoff_intervals = (10, 20, 40, 80)
+        assert (sum(backoff_intervals) <= config.contribution_lambda_timeout)
+        while attempt < len(backoff_intervals):
+            query_job = self._bigquery.query(query)
+            try:
+                query_job.result()
+            except Forbidden as e:
+                if 'Exceeded rate limits' in e.message:
+                    delay = backoff_intervals[attempt]
+                    attempt += 1
+                    msg = ('Exceeded BigQuery rate limit, attempting explicit retry after %.3f '
+                           'seconds. Retry %s of %s.')
+                    log.warning(msg, delay, attempt, len(backoff_intervals), exc_info=e)
+                    time.sleep(delay)
+                else:
+                    raise e
+            else:
+                return query_job
+        raise Exception(f'Unable to complete requested query: {query!r}. '
+                        f'Attempted to retry {len(backoff_intervals)} times.')
```

Comment:
If the query is very long, it may be difficult to correlate the first part of this message with the second. I suggest:

```suggestion
        raise RuntimeError(f'Unable to complete query after {len(backoff_intervals)} '
                           f'attempted retries: {query!r}. ')
```